### DISCUSSION TITLE
Support only building language for Internationalized Routing

### DISCUSSION THREAD

**Discussion by JohnDeved** [vercel/next.js#28470] [Ideas]
### Describe the feature you'd like to request

Trying to build an international site with nextjs.
Requirements and SEO reasons force me to support 250 countries and 6 languages.
that is 1500 different locales in "en-GB" format.

nextjs will refuse to build with that amount of locales with the message:
`Error: Received 1500 i18n.locales items which exceeds the max of 100, please reduce the number of items to continue.`

In all honesty, the site really only needs to build 6 locales for each supported language, the content doesn't change.
The country is only important meta since it is an e-commerce online shop.

### Describe the solution you'd like

A setting that lets you only build each page for each language, not each country + lang.
(or some other solution?)

### Describe alternatives you've considered

maybe it would be also ok to have the languages in the i18n config and have a dynamic prefix for the country for all URLs (global base URL?), I hear Uber and Apple do it this way (if that is possible in nextjs)
/en/gb/

another possible solution was to try to rewrite "en-GB" => "en"
trying to do this as a workarround, sadly causes issues with links on the site, these will redirect to "/en"
```js
  async rewrites() {
    return {
      beforeFiles: [
        {
          source: '/:lang(\\w{2})-:country(\\w{2})/:path*',
          destination: '/:lang/:path*',
        },      
      ],
    }
  }
 ```

**Comment by ijjk** [comment]
Closing as a duplicate of https://github.com/vercel/next.js/issues/28157

**Comment by JohnDeved** [comment]
@ijjk can we reopen this? since https://github.com/vercel/next.js/pull/28429 doesn't really solve my issue.
building all 1500 locales would still be not optimal for me
